### ISSUE TITLE
Can't resolve factory to IoC registration on different thread

### ISSUE THREAD

**Issue by LightInjectedVenom** [seesharper/LightInject#277]
Hi Bernhard,

On our Asp.Net Web API project we are using strategy pattern by help of Lightinject Container.
On Asp.Net Web Api project's main process thread we can resolve the dependencies correctly.
We then create a new thread in that web application. On that thread the dependencies do not resolve.

On our strategy pattern contexts we use IEnumerable dependency injection functionality of LightInject IoC. 

What should be the problem related with other threads ? Any ideas or suggestions will be much appreciated.

Thank you.

Erhan & Ogan

**Comment by LightInjectedVenom** [comment]
Dear Bernhard,

To produce a quick (and most probably dirty) fix to our problem we created a new container in constructor of the class where our new Thread is bound to. Then we re-registered all assembly registrations to that container as well. But my insight is telling me there might be a better solution to this issue.

We are looking forward to your ideas and suggestions about this.

Thank you.

Erhan & Ogan

**Comment by seesharper** [comment]
As a general rule of thumb, you should not start new threads yourself in a web application. Threads represents a very limited resource on a web server since you can potentially have to deal with thousands of requests simultaneously and the threadpool will fill up very quickly. That being said, I don't have enough context to answer your question in detail. The best way to resolve this is to come up with a minimal example application that reproduces the error. Put this sample app up here on GitHub and I will take a look at it.

**Comment by LightInjectedVenom** [comment]
Ok. I will try to reproduce the problem on a similar sample application setup and will send it to you.

**Comment by LightInjectedVenom** [comment]
Hi Bernhard. 

I send you an invitation to my bitbucket repository regarding the sample application about this issue.

For your questions regarding the application, please don't hesitate to contact me from ekaluc@gmail.com

To use the application, you need to setup and run web application first. Then you need to run IntegratorXmlPushSimulator windows forms application. I put a button on that windows forms. When you press that button, it will initiate a post to DataServicesController's Push action. From there on you can follow the rest on debug.

My further investigations tell me that WebScope Scope is somehow interfering with resolving registrations from LightInject container from another thread after WebScope is initiated.

But I don't have any idea or knowledge about how to workaround that.

Best regards,

Erhan Kaluc.

**Comment by LightInjectedVenom** [comment]
Hi Bernhard,

Actually I was trying hard to figure out what the problem should be. After many hang bangs to the nearest wall, I actually figured out that giving no lifetime is the only way of creating a new instance (like new operator of C#) without involving any lifetime scope manager. I was thinking that we need to use PerRequest life time to get new instance from LightInject IoC with every request.

By giving no lifetime, LightInject Container resolves the dependencies as it meant to be. Even after webscope is initiated.

Again once in a while **PerRequestLifeTime** confused me again. I once though that PerRequestLifeTime should be related with Web Request scope. Then I realized the correct lifetime kind for this was PerScope lifetime. So what PerRequest lifetime is meant for is beyond my understanding (most probably because of my bad English understanding).

Regards,

Erhan
